src/processor/Package.ts

```diff
@@ -228,4 +230,44 @@ export class Package {
       pullAt(sd.rules, rulesToRemove);
     });
   }
+
+  private suppressUrlAssignmentOnExtensions(): void {
+    // Loop over all profiles and extensions, removing assignment rules on inline extensions
+    [...this.profiles, ...this.extensions].forEach(sd => {
+      const rulesToRemove: number[] = [];
+      sd.rules.forEach(rule => {
+        if (rule instanceof ExportableContainsRule && rule.path.endsWith('extension')) {
+          rule.items.forEach(item => {
+            const assignmentRuleIdx = sd.rules.findIndex(
+              other =>
+                other.path === `${rule.path}[${item.name}].url` &&
+                other instanceof ExportableFixedValueRule &&
+                other.fixedValue === item.name
```

Comment:
I'm not sure about this.  The only time it will be `item.name` is if it is a sub-extension of an `Extension` definition.  Because inline extensions are only used in that case.

Most cases will be something like:
```
* extension contains http://hl7.org/fhir/StructureDefinition/patient-religion named religion
* extension[religion].url = http://hl7.org/fhir/StructureDefinition/patient-religion (exactly)
```
which I think would fail this code.

Note this is different than the Standalone Extension case below; as it's not defining an extension, but rather _using_ one in the context of some other definition.

Reply:
Yes, but when the extension exists on its own and is used in context of some other definition, SUSHI does _not_ automatically infer that second assignment rule. There is no reason to, the assignment of the `url` is indicated on the definition of the extension itself.

At least that is my understanding from looking at the SUSHI code and from testing. I could be missing something here though so let me know if you have an example of SUSHI doing something I'm missing.

Reply:
I think you're right for cases where the extension is not used.  But I think if you constrain something further in the extension, we still get to this case.  For example:
```
Profile: MyPatient
Parent: Patient
* extension contains http://hl7.org/fhir/StructureDefinition/patient-religion named religion 1..1
* extension[religion].valueCodeableConcept = http://foo.org#bar
```
Will produce this differential:
```json
{
  "element": [
    {
      "id": "Patient.extension",
      "path": "Patient.extension",
      "slicing": {
        "discriminator": [
          {
            "type": "value",
            "path": "url"
          }
        ],
        "ordered": false,
        "rules": "open"
      },
      "min": 1
    },
    {
      "id": "Patient.extension:religion",
      "path": "Patient.extension",
      "sliceName": "religion",
      "min": 1,
      "max": "1",
      "type": [
        {
          "code": "Extension",
          "profile": [
            "http://hl7.org/fhir/StructureDefinition/patient-religion"
          ]
        }
      ]
    },
    {
      "id": "Patient.extension:religion.valueCodeableConcept",
      "path": "Patient.extension.valueCodeableConcept",
      "min": 0,
      "max": "1",
      "patternCodeableConcept": {
        "coding": [
          {
            "code": "bar",
            "system": "http://foo.org"
          }
        ]
      }
    }
  ]
}
```
But... now that I think about it,that's an `OnlyRule`, not an `AssignmentRule` -- so hopefully we're handling that in a different optimizer (and if not, we should).

Clearly I should try to avoid doing code reviews after 8pm. ;-)